extensions/standard-processors/processors/SplitText.cpp

```diff
@@ -0,0 +1,378 @@
+    }
+    header_flow = session_.clone(flow_file_, gsl::narrow<int64_t>(header_fragment->fragment_offset), gsl::narrow<int64_t>(header_fragment->fragment_size));
+    if (!header_flow) {
+      logger_->log_error("Failed to clone header flow!");
+      return -1;
+    }
+  }
+
+  while (auto fragment = fragment_generator.readNextFragment()) {
+    size_t fragment_trim_size = split_text_config_.remove_trailing_new_lines ? fragment->endline_size : 0;
+    if (header_flow) {
+      if (fragment->fragment_size - fragment_trim_size == 0) {
+        createHeaderOnlyFragmentFlow(*header_fragment);
+      } else {
+        mergeHeaderAndFragmentFlows(header_flow, *fragment, fragment_trim_size);
+      }
+    } else if (fragment->fragment_size - fragment_trim_size != 0) {
+      createFragmentFlowWithoutHeader(*fragment, fragment_trim_size);
+    }
+  }
+  if (header_flow) {
+    session_.remove(header_flow);
+  }
+  return fragment_generator.getState() == detail::StreamReadState::EndOfStream ? gsl::narrow<int64_t>(flow_file_->getSize()) : -1;
+}
+
+}  // namespace detail
+
+void SplitText::initialize() {
+  setSupportedProperties(Properties);
+  setSupportedRelationships(Relationships);
+}
+
+void SplitText::onSchedule(core::ProcessContext& context, core::ProcessSessionFactory& /*sessionFactory*/) {
+  split_text_config_.line_split_count = utils::getRequiredPropertyOrThrow<uint64_t>(context, LineSplitCount.name);
+  logger_->log_debug("SplitText line split count: {}", split_text_config_.line_split_count);
+  auto max_fragment_data_size_value = context.getProperty<core::DataSizeValue>(MaximumFragmentSize);
+  if (max_fragment_data_size_value) {
+    split_text_config_.maximum_fragment_size = max_fragment_data_size_value->getValue();
+    logger_->log_debug("SplitText maximum fragment size: {}", split_text_config_.maximum_fragment_size.value());
+  }
+  if (split_text_config_.maximum_fragment_size && split_text_config_.maximum_fragment_size.value() == 0) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Maximum Fragment Size cannot be 0!");
+  }
+  if (split_text_config_.line_split_count == 0 && !split_text_config_.maximum_fragment_size) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Line Split Count is set to 0, but Maximum Fragment Size is not set!");
+  }
+  split_text_config_.header_line_count = utils::getRequiredPropertyOrThrow<uint64_t>(context, HeaderLineCount.name);
+  logger_->log_debug("SplitText header line count: {}", split_text_config_.header_line_count);
+  split_text_config_.header_line_marker_characters = context.getProperty(HeaderLineMarkerCharacters);
+  if (split_text_config_.header_line_marker_characters && split_text_config_.header_line_marker_characters->size() >= detail::SPLIT_TEXT_BUFFER_SIZE) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, fmt::format("SplitText header line marker characters length is larger than the maximum allowed: {}", detail::SPLIT_TEXT_BUFFER_SIZE));
```

Comment:
The error message could (and should IMO) include both the actual set size and the max size, e.g.
```
SplitText header line marker characters length is larger than the maximum allowed: 12345 > 8192
```